src/stack.ts

```diff
@@ -93,7 +94,24 @@ export class App
     private appProps?: cdk.AppProps;
 
     constructor(id: string, createFunc: (scope: App) => void | AppOutputs, props?: AppResourceOptions) {
-        super('cdk:index:App', id, props?.appOptions, props);
+        // This matches the logic found in aws-native. If all of these are undefined the provider
+        // will throw an error
+        const region = native.config.region ?? process.env.AWS_REGION ?? process.env.AWS_DEFAULT_REGION!;
+        // If the user has not provided any providers, we will create an aws-native one by default in order
+        // to enable the autoNaming feature.
+        const providers = props?.providers ?? [
```

Comment:
This could be a bit more precise maybe? If props resets the classic provider, it might still be able to merge in the config for the native provider? Probably a rare use case though, feel free to disregard.